controllers/access_token_mapper.go

```diff
@@ -49,7 +50,14 @@ func (at AccessTokenMapper) toSecretType(secretType corev1.SecretType) map[strin
 	case corev1.SecretTypeDockercfg:
 		ret[corev1.DockerConfigKey] = at.Token
 	case corev1.SecretTypeDockerConfigJson:
-		ret[corev1.DockerConfigJsonKey] = at.Token
+		url := at.ServiceProviderUrl
+		if split := strings.Split(at.ServiceProviderUrl, "://"); len(split) > 1 {
```

Comment:
I think @metlos meant to use `url.Parse` and then just `parsed.Host`. Something like here https://github.com/redhat-appstudio/service-provider-integration-operator/blob/main/pkg/serviceprovider/lookup.go#L50